shared/src/test/diff/gadt/AliasGADT.mls

```diff
@@ -5,31 +5,54 @@ type Test[T] = LitInt | LitStr
 class LitInt(n: Int) { type T = Int }
 class LitStr(s: Str) { type T = Str }
 //│ type Test[T] = LitInt | LitStr
-//│ class LitInt(n: Int) {
-//│   type T = Int
-//│ }
-//│ class LitStr(s: Str) {
-//│   type T = Str
-//│ }
+//│ class LitInt(n: Int)
+//│ class LitStr(s: Str)
 
-// TODO support
 fun foo[T](x: Test[T]): x.T = if x is
   LitInt(n) then n
   LitStr(s) then s
-//│ ╔══[ERROR] Access to type alias member not yet supported
-//│ ║  l.16: 	fun foo[T](x: Test[T]): x.T = if x is
-//│ ╙──      	                         ^^
-//│ ╔══[ERROR] Type mismatch in type selection:
-//│ ║  l.16: 	fun foo[T](x: Test[T]): x.T = if x is
-//│ ║        	                         ^^
-//│ ╙── type `error` cannot be reassigned
-//│ ╔══[ERROR] Access to type alias member not yet supported
-//│ ║  l.16: 	fun foo[T](x: Test[T]): x.T = if x is
-//│ ╙──      	                         ^^
-//│ ╔══[ERROR] Type mismatch in type selection:
-//│ ║  l.16: 	fun foo[T](x: Test[T]): x.T = if x is
-//│ ║        	                         ^^
-//│ ╙── type `error` cannot be reassigned
-//│ fun foo: forall 'T. (x: Test['T]) -> error
+//│ fun foo: forall 'T. (x: Test['T]) -> (Int | Str)
 
+// FIXME
+fun foo[T](x: Test[T]): T = if x is
+  LitInt(n) then n : x.T
+  LitStr(s) then s : x.T
+//│ ╔══[ERROR] Type mismatch in type ascription:
+//│ ║  l.18: 	  LitInt(n) then n : x.T
+//│ ║        	                 ^
+//│ ╟── type `Int` does not match type `T`
+//│ ║  l.5: 	class LitInt(n: Int) { type T = Int }
+//│ ║       	                                ^^^
+//│ ╟── Note: constraint arises from method type parameter:
+//│ ║  l.17: 	fun foo[T](x: Test[T]): T = if x is
+//│ ╙──      	        ^
+//│ ╔══[ERROR] Type mismatch in type ascription:
+//│ ║  l.19: 	  LitStr(s) then s : x.T
+//│ ║        	                 ^
+//│ ╟── type `Str` does not match type `T`
+//│ ║  l.6: 	class LitStr(s: Str) { type T = Str }
+//│ ║       	                                ^^^
+//│ ╟── Note: constraint arises from method type parameter:
+//│ ║  l.17: 	fun foo[T](x: Test[T]): T = if x is
+//│ ╙──      	        ^
+//│ fun foo: forall 'T. (x: Test['T]) -> 'T
 
+// * somehow we need to find a way to associate T in Foo and T in MkF
```

Comment:
Right. It should probably be `type Foo[T0] = MkF & { type T = T0 }`.

(Or in the future, using a type refinement syntax, `MkF { T = T0 }`.)

Reply:
Problem is we currently don't support parsing `{ type T = T0 }`.

Reply:
So maybe let's not try to implement this in the current PR. It would require too many additional changes/features.